saleor/webhook/response_schemas/annotations.py

```diff
@@ -0,0 +1,27 @@
+from typing import Annotated, Any, TypeVar
+
+from pydantic import (
+    BeforeValidator,
+)
+from pydantic_core import PydanticUseDefault
+
+from ...core.utils.metadata_manager import metadata_is_valid
+
+
+def skip_invalid_metadata(value: Any) -> Any:
+    if not metadata_is_valid(value):
+        raise PydanticUseDefault()
+    return value
```

Comment:
this is not correct typing, instead can be type safe with generic, try this:

```python
M = TypeVar("M")


def skip_invalid_metadata(value: M) -> M:
    if not metadata_is_valid(value):
        raise PydanticUseDefault()
    return value
```

Reply:
I will take care of that into separate PR

Reply:
Ok. I'd also recommend to use reasoanble names for type vars, I think "T" is only for example (like `(for i)` ) but is not actually something readable. Maybe it's better to call it with semantics like normal variables, e.g. `MaybeMetadata` or something like this